### PR TITLE
Reorganizing `set -x` calls for verbose execution

### LINKED ISSUE #754: Support extra-verbose output from `llvm-krun`
Currently, `llvm-krun -v` only gives you a limited view of what's going on internally. If, for example, you're debugging an issue `kore-strip` or the other binary tools, the only way to get a hold of the commands being run is to patch `set -ex` into them.

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
Fixes #754 

A simple PR  to improve the `--verbose` use on `llvm-krun` by outputting more steps of the execution process!

### CODE REVIEW

**Review by Baltoli — CHANGES_REQUESTED** [review]
This wasn't quite the intent of the original issue; my point was that the existing verbose mode is usually pretty much what you need to get a reproducible input example from the backend (by copying intermediate files in conjunction with `-save-temps`). However, where it falls down is when you're doing internal developer debugging on something like `kore-strip`. In these cases, we should add a flag `-vv` (or similar) that prints in greater detail the commands that are executed.

Additionally, I recommend reviewing your understanding of bash semantics - as written, this PR appears to move `set -x` calls outside of their original subshell contexts, and adds some in the root shell generally.

**Review by Baltoli — CHANGES_REQUESTED** [review]
This wasn't quite the intent of the original issue; my point was that the existing verbose mode is usually pretty much what you need to get a reproducible input example from the backend (by copying intermediate files in conjunction with `-save-temps`). However, where it falls down is when you're doing internal developer debugging on something like `kore-strip`. In these cases, we should add a flag `-vv` (or similar) that prints in greater detail the commands that are executed.

Additionally, I recommend reviewing your understanding of bash semantics - as written, this PR appears to move `set -x` calls outside of their original subshell contexts, and adds some in the root shell generally.

File: bin/llvm-krun [hunk 1]
```diff
@@ -298,40 +326,76 @@ HERE
...
     echo "Lbl'Stop'Map{}()" | $convert_term >> "$input_file"
+    )
   fi
 
   for param in "${params[@]}"; do
     var_name="params_$param"
     file_name="file_$param"
     binary_file_name="binary_$param"
 
+    set +e
```
Comment (dwightguth) [inline bin/llvm-krun:342]: Why was this line added?

File: bin/llvm-krun [hunk 2]
```diff
@@ -156,8 +161,18 @@ do
     shift; shift
     ;;
 
+    -vvv)
+    verbose+=3
```
Comment (Baltoli) [inline bin/llvm-krun:165]: This line doesn't do what you think it does; this will concatenate the string `3` onto the _string_ value currently stored in `verbose`. For example:

**`test.sh`**
```bash
#!/usr/bin/env bash
verbose=0
verbose+=1
echo $verbose
```

```console
$ ./test.sh
01
```

The correct solution is to use shell [arithmetic expansion](https://tldp.org/LDP/abs/html/arithexp.html)

File: bin/llvm-krun [hunk 3]
```diff
@@ -156,8 +161,18 @@ do
...
+    shift;
```
Comment (Baltoli) [inline bin/llvm-krun:166]: You don't need the semicolon here
```suggestion
    shift
```

**Review by Baltoli — CHANGES_REQUESTED** [review]
Mostly looks good but a couple of small changes needed - thanks @Robertorosmaninho !

File: bin/llvm-krun (hunk 1 above)
Comment (Robertorosmaninho) [reply bin/llvm-krun:342]: my bad

File: bin/llvm-krun [hunk 4]
```diff
@@ -67,7 +68,11 @@ Mandatory arguments to long options are mandatory for short options too.
       --depth INT          Execute up to INT steps
   -i, --initializer INIT   Use INIT as the top cell initializer 
   -nm, --no-expand-macros  Don't expand macros in initial configuration
-  -v, --verbose            Print commands executed to standard error
+  -v, --verbose            Print major commands executed to standard error
+  -vv                      Print also commands executed by intermediate tools to
+                           standard error
```
[the remark this answers is not in the document]
Comment (Robertorosmaninho) [reply bin/llvm-krun:73]: Thanks

File: bin/llvm-krun (hunk 2 above)
Comment (Robertorosmaninho) [reply bin/llvm-krun:165]: My bad, so sorry!

File: bin/llvm-krun (hunk 3 above)
Comment (Robertorosmaninho) [reply bin/llvm-krun:166]: Thanks

File: bin/llvm-krun [hunk 5]
```diff
@@ -156,8 +161,18 @@ do
...
+
+    -vv)
+    verbose=$((verbose+=2))
+    shift
+    ;;
+
     -v|--verbose)
-    verbose=1
+    verbose+=1
     shift;
```
Comment (Baltoli) [inline bin/llvm-krun:176]:
```suggestion
    shift
```

File: bin/llvm-krun [hunk 6]
```diff
@@ -156,8 +161,18 @@ do
...
+    ;;
+
+    -vv)
+    verbose=$((verbose+=2))
+    shift
+    ;;
+
     -v|--verbose)
-    verbose=1
+    verbose+=1
```
Comment (Baltoli) [inline bin/llvm-krun:175]: This line still has the error I reviewed previously

File: bin/llvm-krun [hunk 7]
```diff
@@ -156,8 +161,18 @@ do
     shift; shift
     ;;
 
+    -vvv)
+    verbose=$((verbose+=3))
+    shift
+    ;;
+
+    -vv)
+    verbose=$((verbose+=2))
```
Comment (Baltoli) [inline bin/llvm-krun:170]:
```suggestion
    verbose=$((verbose + 2))
```

File: bin/llvm-krun [hunk 8]
```diff
@@ -156,8 +161,18 @@ do
     shift; shift
     ;;
 
+    -vvv)
+    verbose=$((verbose+=3))
```
Comment (Baltoli) [inline bin/llvm-krun:165]:
```suggestion
    verbose=$((verbose + 3))
```

**Review by Baltoli — CHANGES_REQUESTED** [review]

**Review by Baltoli — APPROVED** [review]
